### ISSUE TITLE
ImportError: cannot import name 'string_types'

### ISSUE THREAD

**Issue by yprokule** [CentOS-PaaS-SIG/linchpin#1613]
**Describe the bug**
Linchpin execution fails due to `ImportError`:

``` bash
linchpin --template-data @extra.yml -v destroy target_one target_two

Traceback (most recent call last):
  File "/tmp/venv/bin/linchpin", line 5, in <module>
    from linchpin.shell import runcli
  File "/tmp/venv/lib/python3.6/site-packages/linchpin/__init__.py", line 24, in <module>
    from linchpin.hooks.state import State
  File "/tmp/venv/lib/python3.6/site-packages/linchpin/hooks/__init__.py", line 52, in <module>
    import git
  File "/tmp/venv/lib/python3.6/site-packages/git/__init__.py", line 38, in <module>
    from git.exc import *                       # @NoMove @IgnorePep8
  File "/tmp/venv/lib/python3.6/site-packages/git/exc.py", line 9, in <module>
    from git.compat import UnicodeMixin, safe_decode, string_types
  File "/tmp/venv/lib/python3.6/site-packages/git/compat.py", line 21, in <module>
    from gitdb.utils.encoding import (
ImportError: cannot import name 'string_types'
```

**To Reproduce**
Install `linchpin==1.10.0`

**Expected behavior**
Linchpin works with `Libvirt` provider

**OS**
`Red Hat Enterprise Linux release 8.1 (Ootpa)`

**Comment by samvarankashyap** [comment]
Comment: 
Bug not observed in the following environment
Fedora 31 
Python 3.7.6

**Comment by 14rcole** [comment]
@yprokule can you check if gitdb is installed in your venv?

**Comment by yprokule** [comment]
> @yprokule can you check if gitdb is installed in your venv?

@14rcole it's there

```
gitdb==0.6.4
gitdb2==3.0.2
ansible==2.9.0
s3transfer==0.3.3
linchpin==1.10.0
```

**Comment by johnbieren** [comment]
Libvirt broke for me too with 1.10.0. I hit this
```
failed: [localhost] (item=[{'name_separator': '-', 'cpu_mode': 'host-passthrough', 'count': 3, 'uri': 'qemu:///system', 'driver': 'kvm', 'arch': 'x86_64', 'remote_user': 'kni', 'boot_dev': 'hd', 'additional_storage': '42G', 'disk_cache': 'unsafe', 'disk_type': 'virtio_scsi', 'network_bridge': '', 'ssh_key': 'id_rsa', 'cloud_config': {'virt_type': 'cloud-init', 'users': [{'name': 'dci', 'gecos': 'Admin User', 'groups': 'wheel', 'sudo': 'ALL=(ALL) NOPASSWD:ALL', 'lock_passwd': False}], 'run_commands': []}, 'networks': [{'name': 'baremetal'}, {'name': 'provisioning', 'dhcp': False}], 'storage': [], 'domain': 'example.com', 'role': 'libvirt_node', 'name': 'master', 'image_src': 'https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2', 'memory': 32768, 'vcpus': 16}, 2, '/var/lib/libvirt/images/linchpin/', 'qcow2', '-', 'unsafe', 'host-passthrough', 'virtio_scsi']) => {"ansible_loop_var": "definition", "changed": false, "definition": [{"additional_storage": "42G", "arch": "x86_64", "boot_dev": "hd", "cloud_config": {"run_commands": [], "users": [{"gecos": "Admin User", "groups": "wheel", "lock_passwd": false, "name": "dci", "sudo": "ALL=(ALL) NOPASSWD:ALL"}], "virt_type": "cloud-init"}, "count": 3, "cpu_mode": "host-passthrough", "disk_cache": "unsafe", "disk_type": "virtio_scsi", "domain": "example.com", "driver": "kvm", "image_src": "https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2", "memory": 32768, "name": "master", "name_separator": "-", "network_bridge": "", "networks": [{"name": "baremetal"}, {"dhcp": false, "name": "provisioning"}], "remote_user": "kni", "role": "libvirt_node", "ssh_key": "id_rsa", "storage": [], "uri": "qemu:///system", "vcpus": 16}, 2, "/var/lib/libvirt/images/linchpin/", "qcow2", "-", "unsafe", "host-passthrough", "virtio_scsi"], "msg": "AnsibleUndefinedVariable: 'cloud_config_used' is undefined"}
```
Is there CI testing for libvirt up in place?

**Comment by 14rcole** [comment]
@yprokule I tried to replicate this with one of the libvirt examples in a RHEL8 container and couldn't.  Can you send me your PinFile?

**Comment by 14rcole** [comment]
@johnbieren I believe so.  I think I know what's going on there though, I'll submit a patch

**Comment by johnbieren** [comment]
https://github.com/gitpython-developers/GitPython/issues/983
switching to gitdb2==3.0.1 got rid of the string_types error for me

**Comment by yprokule** [comment]
> [gitpython-developers/GitPython#983](https://github.com/gitpython-developers/GitPython/issues/983)
> switching to gitdb2==3.0.1 got rid of the string_types error for me

That worked for me as well

**Comment by samvarankashyap** [comment]
Fixed by #1626

**Comment by samvarankashyap** [comment]
fixed by #1621
